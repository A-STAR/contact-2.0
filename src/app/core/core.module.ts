import { ErrorHandler, NgModule, Optional, SkipSelf } from '@angular/core';
import { DatePipe } from '@angular/common';

import { JwtHelper } from 'angular2-jwt';
import { TranslateService } from '@ngx-translate/core';
import { StoreModule } from '@ngrx/store';
import { EffectsModule } from '@ngrx/effects';
import { StoreDevtoolsModule } from '@ngrx/store-devtools';

import { environment } from '../../environments/environment';

import { AuthEffects } from './auth/auth.effects';
import { AuthHttpService } from './auth/auth-http.service';
import { AuthService } from './auth/auth.service';
import { ContentTabService } from '../shared/components/content-tabstrip/tab/content-tab.service';
import { DataService } from './data/data.service';
import { DictionariesEffects } from './dictionaries/dictionaries.effects';
import { DictionariesService } from './dictionaries/dictionaries.service';
import { EntityTranslationsService } from './entity/translations/entity-translations.service';
import { ErrorHandlerService } from './error/error-handler.service';
import { LookupEffects } from './lookup/lookup.effects';
import { LookupService } from './lookup/lookup.service';
import { MenuService } from './menu/menu.service';
import { MetadataService } from './metadata/metadata.service';
import { NotificationsEffects } from './notifications/notifications.effects';
import { NotificationsService } from './notifications/notifications.service';
import { SettingsService } from './settings/settings.service';
import { ThemesService } from './themes/themes.service';
import { throwIfAlreadyLoaded } from './module-import-guard';
import { MetadataEffects } from './metadata/metadata.effects';
import { UserConstantsEffects } from './user/constants/user-constants.effects';
import { UserConstantsService } from './user/constants/user-constants.service';
import { UserDictionariesEffects } from './user/dictionaries/user-dictionaries.effects';
import { UserDictionariesService } from './user/dictionaries/user-dictionaries.service';
import { UserLanguagesEffects } from './user/languages/user-languages.effects';
import { UserLanguagesService } from './user/languages/user-languages.service';
import { UserPermissionsEffects } from './user/permissions/user-permissions.effects';
import { UserPermissionsService } from './user/permissions/user-permissions.service';
import { ValueConverterService } from './converter/value/value-converter.service';

import { rootReducer } from './state/root.reducer';

@NgModule({
  imports: [
    StoreModule.provideStore(rootReducer),
    EffectsModule.run(AuthEffects),
    EffectsModule.run(DictionariesEffects),
    EffectsModule.run(LookupEffects),
    EffectsModule.run(NotificationsEffects),
    EffectsModule.run(UserConstantsEffects),
    EffectsModule.run(UserDictionariesEffects),
    EffectsModule.run(UserLanguagesEffects),
    EffectsModule.run(UserPermissionsEffects),
    EffectsModule.run(MetadataEffects),
    environment.production
      ? []
      : StoreDevtoolsModule.instrumentOnlyWithExtension({
          maxAge: 1024
        })
  ],
  providers: [
    AuthHttpService,
    AuthService,
    ContentTabService,
    EntityTranslationsService,
    DictionariesService,
    DatePipe,
    DataService,
    JwtHelper,
    LookupService,
    MenuService,
    MetadataService,
    NotificationsService,
    SettingsService,
    ThemesService,
    TranslateService,
    UserConstantsService,
    UserDictionariesService,
    UserLanguagesService,
    UserPermissionsService,
    ValueConverterService,
    {
      provide: ErrorHandler,
      useClass: ErrorHandlerService
    }
  ],
})
export class CoreModule {
  constructor(@Optional() @SkipSelf() parentModule: CoreModule) {
    throwIfAlreadyLoaded(parentModule, 'CoreModule');
  }
}
