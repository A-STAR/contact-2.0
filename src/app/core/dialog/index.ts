export abstract class DialogFunctions {
  /**
   * NOTE: this prop cannot come from the prototype and has to be implemented in the children
   *
   * @abstract
   * @type {string}
   * @memberof DialogFunctions
   */
  abstract dialog: string;

  isDialog(dialog: string): boolean {
    return this.dialog === dialog;
  }

  setDialog(dialog: string = null): void {
    this.dialog = dialog;
  }

  closeDialog(): void {
    this.setDialog();
  }
  /**
   * Same as `closeDialog` just to maintain the `on` prefix used in EventEmitters
   *
   * @memberof DialogFunctions
   */
  onCloseDialog(): void {
    this.setDialog();
  }

  onResult(result: any): void {

  }
}
