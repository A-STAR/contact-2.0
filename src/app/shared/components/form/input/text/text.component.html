<label class="m0">
  <div *ngIf="label" class="form-label">
    {{ label | translate }}
    <b *ngIf="required" class="text-danger">*</b>
  </div>
  <div>
    <input
      class="form-control"
      type="text"
      [disabled]="disabled"
      [ngModel]="value"
      (focusout)="onFocusOut()"
      (ngModelChange)="onChange($event)"
    />
  </div>
  <div>
    <span *ngIf="errors">{{ errors | json }}</span>
  </div>
</label>
