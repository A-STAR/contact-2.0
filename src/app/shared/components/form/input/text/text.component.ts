import {
  ChangeDetectionStrategy, ChangeDetectorRef, Component, forwardRef,
  Input, ViewChild, ElementRef, AfterViewInit
} from '@angular/core';
import { ControlValueAccessor, /* NG_VALIDATORS, */ NG_VALUE_ACCESSOR /*, Validator */ } from '@angular/forms';

@Component({
  changeDetection: ChangeDetectionStrategy.OnPush,
  providers: [
    {
      provide: NG_VALUE_ACCESSOR,
      useExisting: forwardRef(() => TextComponent),
      multi: true,
    },
    // {
    //   provide: NG_VALIDATORS,
    //   useExisting: forwardRef(() => TextComponent),
    //   multi: true,
    // },
  ],
  selector: 'app-text',
  templateUrl: 'text.component.html'
})
export class TextComponent implements ControlValueAccessor, /* Validator, */ AfterViewInit {
  @ViewChild('input') inputEl: ElementRef;

  @Input() errors: any;
  @Input() label: string;
  @Input() minLength: number;
  @Input() maxLength: number;
  @Input() required = false;
  @Input() autofocus = false;

  disabled = false;
  value: string;

  constructor(
    private cdRef: ChangeDetectorRef,
  ) {}

  ngAfterViewInit(): void {
    if (this.autofocus) {
      this.inputEl.nativeElement.focus();
    }
  }

  writeValue(value: string): void {
    this.value = value;
    this.cdRef.markForCheck();
  }

  registerOnChange(fn: Function): void {
    this.propagateChange = fn;
  }

  registerOnTouched(fn: Function): void {
    this.propagateTouch = fn;
  }

  setDisabledState(disabled: boolean): void {
    this.disabled = disabled;
  }

  // validate(): any {
  //   switch (true) {
  //     case this.value == null && this.required:
  //       return { required: true };
  //     case this.minLength && this.value && this.value.length < this.minLength:
  //       return { min: { minLength: this.minLength } };
  //     case this.maxLength && this.value && this.value.length > this.maxLength:
  //       return { max: { maxLength: this.maxLength } };
  //     default:
  //       return null;
  //   }
  // }

  onChange(value: string): void {
    this.value = value;
    this.propagateChange(value);
    this.cdRef.markForCheck();
  }

  onFocusOut(): void {
    this.propagateTouch();
  }

  private propagateChange: Function = () => {};
  private propagateTouch: Function = () => {};
}
