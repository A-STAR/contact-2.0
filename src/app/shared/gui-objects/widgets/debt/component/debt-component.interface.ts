export interface IDebtComponent {
  id: number;
  typeCode: number;
  sum: number;
  currencyId: number;
}
