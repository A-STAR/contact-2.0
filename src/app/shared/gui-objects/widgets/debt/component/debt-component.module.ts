import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

import { DebtComponentGridModule } from './grid/debt-component-grid.module';

import { DebtComponentService } from './debt-component.service';

@NgModule({
  imports: [
    DebtComponentGridModule,
    CommonModule,
  ],
  exports: [
    DebtComponentGridModule,
  ],
  providers: [
    DebtComponentService,
  ]
})
export class DebtComponentModule { }
