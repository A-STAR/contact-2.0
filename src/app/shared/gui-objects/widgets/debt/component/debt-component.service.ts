import { Injectable } from '@angular/core';
import { Observable } from 'rxjs/Observable';

import { IDebtComponent } from './debt-component.interface';

import { DataService } from '../../../../../core/data/data.service';
import { NotificationsService } from '../../../../../core/notifications/notifications.service';

@Injectable()
export class DebtComponentService {
  static MESSAGE_DEBT_COMPONENT_SAVED = 'MESSAGE_DEBT_COMPONENT_SAVED';

  constructor(
    private dataService: DataService,
    private notificationsService: NotificationsService,
  ) {}

  fetchAll(debtId: number): Observable<Array<IDebtComponent>> {
    return this.dataService
      .read('/debts/{debtId}/components', { debtId })
      .map(response => response.components)
      .catch(this.notificationsService.error('errors.default.read').entity('entities.debtComponents.gen.plural').dispatchCallback());
  }

  fetch(debtId: number, debtComponentId: number): Observable<IDebtComponent> {
    return this.dataService
      .read('/debts/{debtId}/components/{debtComponentId}', { debtId, debtComponentId })
      .map(response => response.components[0])
      .catch(this.notificationsService.error('errors.default.read').entity('entities.debtComponents.gen.singular').dispatchCallback());
  }

  create(debtId: number, debtComponent: IDebtComponent): Observable<void> {
    return this.dataService
      .create('/debts/{debtId}/components', { debtId }, debtComponent)
      .catch(this.notificationsService.error('errors.default.create').entity('entities.debtComponents.gen.singular').dispatchCallback());
  }

  update(debtId: number, debtComponentId: number, debtComponent: Partial<IDebtComponent>): Observable<void> {
    return this.dataService
      .update('/debts/{debtId}/components/{debtComponentId}', { debtId, debtComponentId }, debtComponent)
      .catch(this.notificationsService.error('errors.default.update').entity('entities.debtComponents.gen.singular').dispatchCallback());
  }

  delete(debtId: number, debtComponentId: number): Observable<void> {
    return this.dataService
      .delete('/debts/{debtId}/components/{debtComponentId}', { debtId, debtComponentId })
      .catch(this.notificationsService.error('errors.default.delete').entity('entities.debtComponents.gen.singular').dispatchCallback());
  }
}
