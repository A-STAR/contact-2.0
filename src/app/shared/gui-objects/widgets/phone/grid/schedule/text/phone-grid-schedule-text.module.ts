import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

import { DynamicFormModule } from '../../../../../../components/form/dynamic-form/dynamic-form.module';

import { PhoneGridScheduleTextComponent } from './phone-grid-schedule-text.component';

@NgModule({
  imports: [
    CommonModule,
    DynamicFormModule,
  ],
  exports: [
    PhoneGridScheduleTextComponent,
  ],
  declarations: [
    PhoneGridScheduleTextComponent,
  ],
  entryComponents: [
    PhoneGridScheduleTextComponent,
  ]
})
export class PhoneGridScheduleTextModule { }
