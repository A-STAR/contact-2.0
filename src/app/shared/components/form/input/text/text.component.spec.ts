import { async as Async, ComponentFixture, TestBed } from '@angular/core/testing';
import { By } from '@angular/platform-browser';
import { FormsModule } from '@angular/forms';
import { TranslateLoader, TranslateModule } from '@ngx-translate/core';
import { Observable } from 'rxjs/Observable';
import { of } from 'rxjs/observable/of';

import { TextComponent } from './text.component';

describe('TextComponent', () => {
  let fixture: ComponentFixture<TextComponent>;

  beforeEach(Async(() => {
    TestBed
      .configureTestingModule({
        declarations: [
          TextComponent,
        ],
        imports: [
          FormsModule,
        ]
      })
      .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(TextComponent);
  });

  it('should render', () => {
    fixture.detectChanges();
    expect(fixture.nativeElement).toMatchSnapshot();
  });

  it('should handle losing focus', () => {
    fixture.detectChanges();
    fixture.debugElement.query(By.css('input')).nativeElement.dispatchEvent(new Event('blur'));
    fixture.detectChanges();
    expect(fixture.nativeElement).toMatchSnapshot();
  });

  it('should handle input', () => {
    fixture.detectChanges();
    fixture.debugElement.query(By.css('input')).nativeElement.dispatchEvent(new Event('input'));
    fixture.detectChanges();
    expect(fixture.nativeElement).toMatchSnapshot();
  });
});
