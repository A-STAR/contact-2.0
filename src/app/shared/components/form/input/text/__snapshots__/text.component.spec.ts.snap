// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextComponent should handle input 1`] = `
<div
  id="root2"
  ng-version="5.2.4"
>
  <input
    class="form-control ng-untouched ng-valid ng-dirty"
    ng-reflect-is-disabled="false"
    ng-reflect-model=""
    type="text"
  />
  

</div>
`;

exports[`TextComponent should handle losing focus 1`] = `
<div
  id="root1"
  ng-version="5.2.4"
>
  <input
    class="form-control ng-pristine ng-valid ng-touched"
    ng-reflect-is-disabled="false"
    type="text"
  />
  

</div>
`;

exports[`TextComponent should render 1`] = `
<div
  id="root0"
  ng-version="5.2.4"
>
  <input
    class="form-control ng-untouched ng-pristine ng-valid"
    ng-reflect-is-disabled="false"
    type="text"
  />
  

</div>
`;
